### PR TITLE
[TTAHUB-2245] Communication log backend

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Add backend for communication log

## How to test

No UI yet, so just confirm that it makes sense based on the spec (blob implementation) and is adequately tested

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2245

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/routes/communicationLog/handlers.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  } catch (error) {
+    await handleErrors(req, res, error, logContext);
+  }
+};
+
+const updateLogById = async (req: Request, res: Response) => {
+  try {
+    const policy = await getAuthorizationByLogId(req, res);
+    if (!policy.canUpdateLog()) {
+      res.status(httpCodes.FORBIDDEN).send();
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply src/routes/communicationLog/handlers.ts:72]: thanks

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good overall will do testing once hooked up to UI.

File: src/services/communicationLog.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  const logs = await CommunicationLog.findAll({
+    where: {
+      recipientId,
+      [Op.and]: [
+        ...scopes,
+      ],
+    },
+  });
+  return logs;
+};
```
Comment (GarrettEHill) [inline src/services/communicationLog.ts:30]: just a question, why not just call the findAll directly instead of storing it to a var first?

**Review by GarrettEHill — APPROVED** [review]

File: src/services/communicationLog.ts (hunk 2 above)
Comment (thewatermethod) [reply src/services/communicationLog.ts:30]: mild brain damage

**Pull request merged**